lib/flow/VaporField.cpp

```diff
@@ -0,0 +1,424 @@
+        if( time < _timestamps.front() || time > _timestamps.back() )
+            return false;
+
+        // Then locate the 2 time steps
+        size_t floor;
+        int rv  = LocateTimestamp( time, floor );
+        if( rv != 0 ) return false;
+
+        // Second test if pos is inside of time step "floor"
+        for( auto v : VelocityNames )   // cannot use references...
+            if( !v.empty() )
+            {
+                rv = _getAGrid( floor, v, &grid );
+                assert( rv == 0 );
+                if( !grid->InsideGrid( coords ) )
+                    return false;
+            }
+
+        // If time is larger than _timestamps[floor], we also need to test _timestamps[floor+1]
+        if( time > _timestamps[floor] )
+        {
+            for( auto v : VelocityNames )   // cannot use references
+                if( !v.empty() )
+                {
+                    rv = _getAGrid( floor + 1, v, &grid );
+                    assert( rv == 0 );
+                    if( !grid->InsideGrid( coords ) )
+                        return false;
+                }
+        }
+    }
+
+    return true;
+}
+
+
+bool
+VaporField::InsideVolumeScalar( float time, const glm::vec3& pos )
+{
+    if( ScalarName.empty() )
+        return NO_FIELD_YET;
+    std::string scalarname = ScalarName;    // const requirement...
+    const std::vector<double> coords{ pos.x, pos.y, pos.z };
+    const VAPoR::Grid* grid = nullptr;
+    assert( _isReady() );
+
+    // In case of steady field, we only check a specific time step
+    if( IsSteady )
+    {
+        size_t currentTS = _params->GetCurrentTimestep();
+        int rv = _getAGrid( currentTS, scalarname, &grid );
+        assert( rv == 0 );
+        if( !grid->InsideGrid( coords ) )
+            return false;
+    }
+    else    // we check two time steps
+    {
+        // First check if the query time is within range
+        if( time < _timestamps.front() || time > _timestamps.back() )
+            return false;
+
+        // Then locate the 2 time steps
+        size_t floor;
+        int rv  = LocateTimestamp( time, floor );
+        if( rv != 0 ) return false;
+
+        // Second test if pos is inside of time step "floor"
+        rv = _getAGrid( floor, scalarname, &grid );
+        assert( rv == 0 );
+        if( !grid->InsideGrid( coords ) )
+            return false;
+
+        // If time is larger than _timestamps[floor], we also need to test _timestamps[floor+1]
+        if( time > _timestamps[floor] )
+        {
+            rv = _getAGrid( floor + 1, scalarname, &grid );
+            assert( rv == 0 );
+            if( !grid->InsideGrid( coords ) )
+                return false;
+        }
+    }
+
+    return true;
+}
+
+void
+VaporField::GetFirstStepVelocityIntersection( glm::vec3& minxyz, glm::vec3& maxxyz )
+{
+    const VAPoR::Grid* grid = nullptr;
+    std::vector<double> min[3], max[3];
+
+    for( int i = 0; i < 3; i++ )
```

Comment:
hard wired loop limits aren't so great. Better to loop over VelocitNames.size(). What happens with a 2D field?

Reply:
Hmm, the variable `VelocityNames` is declared as an array of 3, but not an std container. I believe this is a case where we could use hard wired loop limits.

Reply:
Nevertheless, the application crashes when I change one of the variables in a 3D data set to "no variable". This may be part of the problem.

Reply:
Let's mark that crash as a bug, and I'm pretty sure it isn't caused by this loop. 
(Given how this variable is declared: `std::string VelocityNames[3];` it'll always have three elements.)